server/model/base.go

```diff
@@ -0,0 +1,36 @@
+package model
+
+import (
+	"errors"
+
+	mmModel "github.com/mattermost/mattermost/server/public/model"
+)
+
+const (
+	IdLength = 27
+)
+
+var (
+	errEmptyId   = errors.New("ID cannot be empty")
+	errInvalidId = errors.New("invalid ID")
+)
+
+func IsValidId(id string) error {
+	if id == "" {
+		return errEmptyId
+	}
+
+	// ID should have the right length
+	if len(id) != IdLength {
+		return errInvalidId
+	}
+
+	// ID should have the right format.
+	// Excluding the first letter as it represents the block type
```

Comment:
Probably not for this PR, but we should validate that the letter is one of `utils.IDType*`